### PR TITLE
feat: epic typing indicator sender (WPB-4590)

### PR DESCRIPTION
**Pull request by yamilmedina**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-4590" title="WPB-4590" target="_blank">TaskWPB-4590</a>  Sending conversation.typing events
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like `SQPIT-764`
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

Send typing indicators events to conversation.

Needs releases with:

- https://github.com/wireapp/kalium/pull/2127

- [x] GitHub link to other pull request

#### Test Coverage (Optional)

- [x] I have added automated test to this contribution

### Attachments (Optional)

https://github.com/wireapp/wire-android-reloaded/assets/5806454/24b3d724-6a48-431b-b154-3b7fcf64095a

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [x] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1241](https://10.10.124.134/job/AR-build-pipeline/1241/) **failed**.

**Comment by AndroidBob** [comment]
Build [1242](https://10.10.124.134/job/AR-build-pipeline/1242/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.5.0(7806415)(7806415)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.5.0(7806415)(7806415)-staging-compat.apk)

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/UsersTypingIndicator.kt [hunk 1]
```diff
@@ -137,20 +135,31 @@ private fun HorizontalBouncingWritingPen(
 ) {
     Row(modifier = Modifier.fillMaxHeight()) {
         val position by infiniteTransition.animateFloat(
-            initialValue = -5f, targetValue = -1f,
+            initialValue = -10f, targetValue = -2f,
             animationSpec = infiniteRepeatable(
-                animation = tween(1_000, easing = FastOutSlowInEasing),
+                animation = tween(1_500, easing = FastOutSlowInEasing),
```
Comment (gongracr) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/UsersTypingIndicator.kt]: Do we want to extract these values to some `Dimens` variable for easier concurrent modification?

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/UsersTypingIndicator.kt [hunk 2]
```diff
@@ -137,20 +135,31 @@ private fun HorizontalBouncingWritingPen(
...
                 repeatMode = RepeatMode.Reverse
             ),
             label = infiniteTransition.label
         )
 
+        val iconAlpha: Float by infiniteTransition.animateFloat(
+            initialValue = 0.1f,
+            targetValue = 1f,
+            animationSpec = infiniteRepeatable(
+                animation = tween(durationMillis = 1_500, easing = FastOutLinearInEasing),
```
Comment (gongracr) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/UsersTypingIndicator.kt]: same than above

**Review by gongracr — APPROVED** [review]
Real nice ✨ and extra bonus points for the nice tests 🧪 💪

**Review by vitorhugods — APPROVED** [review]

**Review by alexandreferris — APPROVED** [review]

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/UsersTypingIndicator.kt (hunk 1 above)
Comment (yamilmedina) [reply app/src/main/kotlin/com/wire/android/ui/home/conversations/UsersTypingIndicator.kt]: Good idea, I'll take a look on how we are doing for some animations, 
I tried to look for dimensions that fit, but didn't find anything so far

**Comment by AndroidBob** [comment]
Build [1249](https://10.10.124.134/job/AR-build-pipeline/1249/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.5.0(7809684)(7809684)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.5.0(7809684)(7809684)-staging-compat.apk)

**Pull request merged**